uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequest.php

```diff
@@ -477,6 +477,14 @@ private static function validateEntitlementAndWorkingDayAndBalanceChange($params
 
     $currentBalance = $leavePeriodEntitlement->getBalance();
 
+    if (!empty($params['id'])) {
+      $oldLeaveRequest = self::findById($params['id']);
+
+      if ($oldLeaveRequest && $oldLeaveRequest->id && self::isAlreadyApproved($oldLeaveRequest)) {
+        $currentBalance = $leavePeriodEntitlement->getBalance([$oldLeaveRequest->id]);
+      }
+    }
+
```

Comment:
What about this?

```php
    $requestsToExcludeFromBalance = [];
    if (!empty($params['id'])) {
      $oldLeaveRequest = self::findById($params['id']);

      if ($oldLeaveRequest && $oldLeaveRequest->id && self::isAlreadyApproved($oldLeaveRequest)) {
         $requestsToExcludeFromBalance[] = $oldLeaveRequest->id;
      }
    }
    $currentBalance = $leavePeriodEntitlement->getBalance();
```

It's a small optimization but with that we avoid calling getBalance() twice when the leave request has already been approved